File: Python/Bubble-Sort-Recursive.py
from random import randint

def is_sorted(arr):
    length = len(arr)

    for x in range(length-1):
        if arr[x] > arr[x+1]:
            return False

    return True


def bubble_sort(arr, length):
    if length <= 1:
        return arr

    for x in range(len(arr)-1):
        if arr[x] > arr[x+1]:
            arr[x], arr[x+1] = arr[x+1], arr[x]

    arr = bubble_sort(arr, length-1)

    return arr


if __name__ == '__main__':
    arr = [randint(0, 100) for _ in range(10)]
    print(f'Array: {arr}. Sorted? {is_sorted(arr)}')
    arr = bubble_sort(arr, len(arr))
    print(f'Array: {arr}. Sorted? {is_sorted(arr)}')

File: Ruby/Bubble_Sort_Recursive.rb
def is_sorted?(arr)
    len = arr.length

    0.upto(len-2) do |x|
        if arr[x] > arr[x+1]
            return false
        end
    end

    return true
end


def bubble_sort(arr, len)
    if len <= 1
        return arr
    end

    0.upto(len -2 ) do |x|
        if arr[x] > arr[x+1]
            arr[x], arr[x+1] = arr[x+1], arr[x]
        end
    end

    arr = bubble_sort(arr, len-1)

    return arr
end


if __FILE__ == $0
    arr = 10.times.map { |x| (Random.rand() * 100).to_i + x }
    puts "Array #{arr}. Sorted? #{is_sorted?(arr)}"
    arr = bubble_sort(arr, arr.length)
    puts "Array #{arr}. Sorted? #{is_sorted?(arr)}"
end
